### ISSUE TITLE
Urza, Academy Headmaster bug

### ISSUE THREAD

**Issue by jeffwadsworth** [magefree/mage#5589]
https://www.slightlymagic.net/forum/viewtopic.php?f=70&t=23164&start=60#p229849
It freezes the game in any mode.

**Comment by Zzooouhh** [comment]
Reposting from Gitter:

Urza, Academy Headmaster really stumps me. The debug shows nothing. Removing all loyalty abilities from him makes him stop freezing (I believed the issue would be located somewhere within the RandomEffect). I've removed all of my UST images, replaced the whole RandomEffect class from him and gave him the most barebones placeholder effect instead (no parameters and apply simply returns true) - the freezes happen again.

So I gave an otherwise blank Urza a DrawCardAllEffect ability, and he's not frozen... but when I try to copy the whole DrawCardAllEffect class into his file and call that effect in his ability (identical to DrawCardAllEffect in all but name), he continues freezing... huh?
